### DISCUSSION TITLE
Binary manipulation in Forth

### DISCUSSION THREAD

**Discussion by ctrilsbach** [tabemann/zeptoforth#74] [Q&A] [answered]
Are there any words to logically shift binary similar to C's >> & << in Forth?

If not would they be difficult to create? Presumably assembler would have these commands.

**Comment by tabemann** [comment, marked as answer]
For single-cell integers there are `lshift`, for shift left, `rshift`, for shift right (i.e unsigned shift right), and `arshift`, for arithmetic shift right (i.e. signed shift right).

For double-cell integers there are `2lshift`, for shift left, and `2rshift`, for shift right (i.e. unsigned shift right). There is also `2arshift`, but I found a significant bug when I was testing it this morning, so it will need more work.

**Reply by ctrilsbach** [reply]
Great thanks Travis.

**Reply by tabemann** [reply]
There is now a fix to `2arshift` in the repository, in both the `master` and `devel` branches.
